### ISSUE TITLE
Loss becomes NaN when response is empty due to `max_prompt_length >= max_length` in SimPO

### ISSUE THREAD

**Issue by vigneshwaran** [huggingface/trl#3339]
Encountered, `NaN` while training using SimPO under condition max_prompt_length == max_length.

**Details**:
```
Algorithm : SimPO
max_prompt_length: 2048
max_length: 2048
```

**Issue**:
The normalization for loss is done using the number of non-pad tokens in the response. If the response is empty, this results in division by zero i.e., `0/0`, causing `loss = NaN`.

**Expected Behavior**:
- TRL should skip or filter out samples with an empty response to avoid `NaN` losses
**OR**
- The implementation should enforce `max_prompt_length < max_length` during data preparation or training.
**OR**
- Clamp denominator to 1.0 when response length is empty to avoid division by zero at [cpo_trainer.py](https://github.com/huggingface/trl/blob/2bc182c4fb8180a3cb815c61ca6c528a86559d14/trl/trainer/cpo_trainer.py#L717)
